### PR TITLE
[Ready for Review] Feature/api/profile pic

### PR DESCRIPTION
**Pull request by samchrisinger**
# Purpose 

brianjgeiger reports:

> We had a brief discussion on this issue, and it's leading to the following proposed changes:
1) v2 serializer uses profile_image_url instead of gravatar_url because we may not always use gravatar;
2) modify the user view to accept a query parameter to set the profile_image_size.
By default, it will still return a specified size.

# Changes

- Update V2 user serializer to use profile_image_url instead of gravatar_url
- Let user list and detail views take profile_image_size query param
  - write tests to document this
- Update User model to have a profile_image_url method, and replace usages of User.gravatar_url property with new method
- Update profile image size constants in settings to have more general names

# Side Effects 

We should track down all gravatar specific references and replace them with the generalized profile_image_url method.

### CODE REVIEW

File: api/users/serializers.py [hunk 1]
```diff
@@ -18,7 +19,12 @@ class UserSerializer(JSONAPISerializer):
     family_name = ser.CharField(required=False, allow_blank=True, help_text='For bibliographic citations')
     suffix = ser.CharField(required=False, allow_blank=True, help_text='For bibliographic citations')
     date_registered = ser.DateTimeField(read_only=True)
-    gravatar_url = ser.URLField(required=False, read_only=True, help_text='URL for the icon used to identify the user. Relies on http://gravatar.com ')
+
+    profile_image_url = ser.SerializerMethodField(required=False, read_only=True)
```
Comment (sloria) [inline api/users/serializers.py]: `required` and `read_only` are unnecessary, since `SerializerMethodFields` are already read-only.

**Comment by sloria** [comment]
This does not seem to address the issue of hardcoded sizes: https://openscience.atlassian.net/browse/OSF-4092 . If the client doesn't pass any query params, there should be no size specified in the gravatar URL. 

@samchrisinger Can you look into this?

**Comment by sloria** [comment]
Will continue CR when the above issue is addressed.

**Comment by samchrisinger** [comment]
@sloria, this is addressed

**Comment by sloria** [comment]
Looks good.

**Comment by sloria** [comment]
Resolved conflicts and merged with develop

**Pull request closed without merging**